src/AzureAuth.Test/CommandMainTest.cs

```diff
@@ -503,6 +509,140 @@ public void TestCacheFileOptionWithWindowsAbsolutePath()
             subject.CacheFilePath.Should().Be(path);
         }
 
+        /// <summary>
+        /// Test to generate event data from a null authflow result.
+        /// </summary>
+        [Test]
+        public void TestGenerateEvent_FromNullAuthResult()
+        {
+            AuthFlowResult authFlowResult = null;
+            var subject = this.serviceProvider.GetService<CommandMain>();
+
+            // Act
+            var eventData = subject.GenerateEventData(authFlowResult);
+
+            // Assert
+            eventData.Should().BeNull();
+        }
+
+        /// <summary>
+        /// Test to generate event data from an authflow result with null token result and null errors.
+        /// </summary>
+        [Test]
+        public void TestGenerateEvent_From_AuthFlowResult_With_Null_TokenResult_Null_Errors()
+        {
+            AuthFlowResult authFlowResult = new AuthFlowResult(null, null, "AuthFlowName");
+            var subject = this.serviceProvider.GetService<CommandMain>();
+
+            // Act
+            var eventData = subject.GenerateEventData(authFlowResult);
+
+            // Assert
+            eventData.Properties.Should().NotContainKey("msal_correlation_ids");
+            eventData.Properties.Should().NotContainKey("error_messages");
+            eventData.Measures.Should().NotContainKey("token_validity_hours");
+            eventData.Properties.Should().NotContainKey("is_silent");
+
+            eventData.Properties.TryGetValue("auth_mode", out string authMode);

```

Comment:
Shouldn't this be `authflow_name` ? This is where tragedy of "AuthMode" enum comes into play. Right now `auth_mode` is an enum that has values that represent different things and we have tried to store too much in it. 

I think we want to refactor how we use auth_mode. Now that we have the AuthFlowName property on the AuthFlow Result itself. All we want on the TokenResult is a bool - `Interactive`. And we should be able to delete the AuthMode enum all together, and instead of setting it to different values, it is only ever set to true or false.

Reply:
Sorry I was referring to "auth Mode" above but was meaning to talk about "Auth Type".